pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -437,11 +449,37 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         }
     }
 
+    protected void checkRateLimit(BaseCommand cmd) {
+        if (cmd.getType() == BaseCommand.Type.PONG && cmd.getType() == BaseCommand.Type.PING) {
+            return;
+        }
+        if (requestRateLimiter.acquire(1) == 0) {
+            log.warn("[{}] Reached rate limitation", this);
+            // Stop receiving requests.
+            pausedDueToRateLimitation = true;
+            ctx.channel().config().setAutoRead(false);
+            // Resume after 1 second.
+            ctx.channel().eventLoop().schedule(() -> {
+                if (pausedDueToRateLimitation) {
+                    log.info("[{}] Resuming connection after rate limitation", this);
+                    ctx.channel().config().setAutoRead(true);
+                    pausedDueToRateLimitation = false;
+                }
+            }, 1, TimeUnit.SECONDS);
+        }
```

Comment:
In my PIP-434, I've explained that auto read shouldn't be toggled directly. You must use  ServerCnxThrottleTracker instead.
The channel must be throttled if any of the conditions for throttling are present.

Reply:
Since ServerCnxThrottleTracker was introduced with pulsar-3.2, to guarantee the PR can be cherry-picked more easily, I will submit a separate PR to do this

Reply:
This PR is targeting master branch. You should use `ServerCnxThrottleTracker` in master branch PRs. A separate backport can be made for Pulsar 3.0 if this PIP is also for Pulsar 3.0.x.